text/0000-introduce empty ranges.md

```diff
@@ -0,0 +1,431 @@
+>           or _inc.nan() or _inc.infinite()
+>       else
+>         false
+>       end
+>     _infinite =
+>       is_float_infinite
+>         or ((_inc == 0) and (min != max))    // no progress
+>         or ((_min < _max) and (_inc < 0)) // progress into other directions
+>         or ((_min > _max) and (_inc > 0)) 
+> ```
+
+## Proposed reclassifications
+
+This RFC partitions the set of currently *infinite* `Range` cases into those now considered *empty*, and a very small set of those that remain *infinite*. To discuss this discrimination, we first list three criteria. All three must be met for a currently *infinite* Range to *not* now be reclassified as empty -- if either of these criteria is violated, the Range is *empty*; if all three are met, a currently *infinite* Range remains so under this proposal.
+
+  **Criterion 1** - progress from `min` to `max` must be possible and necessary. This necessitates the "no-progress expression" (discussion below) to be `false`.
+  
+  **Criterion 2** - the iterator that realizes the progress must produce *finite* values that lie within `[min, max)`.
+  
+  **Criterion 3** - none of the parameters `min, max, step` can be float `NaN`. (discussion below)
+
+### Criterion 1 -- no-progress expression
+
+The first criterion is related to the question whether progress from `min` to `max` is possible or needed. Numerical values including the floating point values `+Inf` and `-Inf` can be meaningfully compared in expressions like `min > max` or `max > min`. Based on this, an obvious absence of *progress* of the iteration from min towards max based on the sign of the Range parameters can be tested by the following *no-progress expression*: `((min <= max) and (step <= 0)) or ((min >= max) and (step >= 0))`, for both finite and infinite bounds.
+
+Besides the cases with obvious sign-incompatibility of the three Range parameters, this expression also covers the cases of equal bounds `min == max` and `step == 0`. Equal bounds correspond to `[N, N)` which cannot contain any value, and the case of `step == 0` to a situation where no progress is possible. In the current implementation of Range, a somewhat similar couple of expressions are used with the comment `// no progress` and `// progress into other directions`, but they are treated as a sufficient condition for *infinite*. Here, this criterion is used to test for empty Ranges because no valid *trajectory* from min to max does exists with the provided parameters.
+
+#### Examples for empty Ranges that violate criterion 1
+
+`Range(0, 10, -1)` is *empty* nd no longer *infinite*, but so is in fact `Range(0, inf, -1)` that is also considered *infinite* currently. While one can generally never produce a complete list of elements that would incrementally move from say 0 to +Inf, we can know in the case of `Range(0, inf, -1)` that the Range must be *empty* because of the incompatible sign of the step parameter relative to the signs of `min` and `max`.
```

Comment:
Typo: `Range(0, 10, -1)` is *empty* and no longer *infinite*

Reply:
What is the typo?

Reply:
"and" is missing an "a"

Reply:
Ah - thx.